OpenDreamRuntime/DreamManager.Connections.cs

```diff
@@ -106,11 +106,16 @@ public sealed partial class DreamManager {
                             var length = BitConverter.ToUInt16(buffer);
 
                             buffer = new byte[length];
-                            var read = await from.ReceiveAsync(buffer, cancellationToken);
-                            if (read != buffer.Length) {
-                                _sawmill.Warning("failed to parse byond topic due to insufficient data read");
-                                return null;
-                            }
+                            var totalRead = 0;
+                            do {
+                                var read = await from.ReceiveAsync(buffer, cancellationToken);
```

Comment:
Won't this write other parts of the packet to the beginning of `buffer`?

Reply:
doy